app/src/main/java/com/example/dogbook/SignupActivity/SignupActivity.java

```diff
@@ -3,15 +3,122 @@
+    CheckBox cbLookingForPlaymates;
+    EditText etOwnerName;
+    EditText etOwnerContact;
+    Button btnSignup;
+    Button btnLogin;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_signup);
+        initView();
+        btnSignup.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                signUp();
+            }
+        });
+
+        btnLogin.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                goLoginActivity();
+            }
+        });
+
+    }
+
+    private void initView() {
+        etUsername = findViewById(R.id.etUsername);
+        etPassword = findViewById(R.id.etPassword);
+        etConfirmPassword = findViewById(R.id.etConfirmPassword);
+        etBreed = findViewById(R.id.etBreed);
+        dpBirthday = findViewById(R.id.dpBirthday);
+        cbLookingForPlaymates = findViewById(R.id.cbLookingForPlaymates);
+        etOwnerName = findViewById(R.id.etOwnerName);
+        etOwnerContact = findViewById(R.id.etOwnerContact);
+        btnSignup = findViewById(R.id.btnSignup);
+        btnLogin = findViewById(R.id.btnLogin);
+    }
+
+    private void signUp() {
+        //Password confirmation was invalid
+        if (!etPassword.getText().toString().equals(etConfirmPassword.getText().toString())) {
+            Toast.makeText(this, "Passwor confirmation was invalid", Toast.LENGTH_SHORT).show();
+            return;
+        }
+        ParseUser user = new ParseUser();
+        user.setUsername(etUsername.getText().toString());
+        user.setPassword(etPassword.getText().toString());
+        user.put("ownerName", etOwnerName.getText().toString());
+        user.put("breed", etBreed.getText().toString());
+        user.put("birthday", getBirthdayDate());
+        user.put("lookingForPlaymates", cbLookingForPlaymates.isChecked());
+        if (!etOwnerContact.getText().toString().isEmpty()) {
+            user.put("ownerContact", etOwnerContact.getText().toString());
+        }
+
+        user.signUpInBackground(new SignUpCallback() {
+            @Override
+            public void done(ParseException e) {
+                //Sign up succeeded
+                if (e == null){
+                    goMainActivity();
+                }
+                //Sign up did NOT succeed
+                else {
+                    //TODO: Show specific issues to user with different toasts
+                    Toast.makeText(SignupActivity.this, "Signup was not possible", Toast.LENGTH_LONG).show();
+                    Log.e(TAG, "Issue with signup", e);
+                }
+            }
+        });
+    }
+
+    private Date getBirthdayDate() {
+        Date date = new Date(dpBirthday.getYear() - 1900, dpBirthday.getMonth(), dpBirthday.getDayOfMonth());
```

Comment:
Let's save 1900 into a static constant to avoid magic numbers.  You can also return right here

```
return new Date(...);
```